src/app/Instance/InstancesListPage/InstancesListPage.tsx

```diff
@@ -193,34 +134,54 @@ const InstancesListPage = (): JSX.Element => {
         </TextContent>
       </PageSection>
       <PageSection>
-        <TableWithPagination
-          columns={columnNames}
-          customToolbarElement={
-            <React.Fragment>
-              <Button
-                ouiaId="create-smart-event-instance"
-                onClick={(): void => setShowCreateInstance(true)}
-              >
-                {t("instance.createSEInstance")}
-              </Button>
-              <CreateInstance
-                isLoading={false}
-                isModalOpen={showCreateInstance}
-                onClose={(): void => setShowCreateInstance(false)}
-                onCreate={(): void => setShowCreateInstance(false)}
-              />
-            </React.Fragment>
-          }
-          onDetailsClick={(rowData): void => {
-            setSelectedInstance(rowData as unknown as Instance);
-            setShowInstanceDrawer(true);
-          }}
-          rows={instances}
-          tableLabel={t("openbridgeTempDictionary:instance.instancesListTable")}
-        />
+        {isFirstLoad && isLoading && (
+          <TableWithPaginationSkeleton
+            columns={columnNames}
+            customToolbarElement={customToolbarElement}
+            totalRows={currentPageSize}
+            hasActionColumn={true}
+          />
+        )}
+        {bridgeListResponse?.items && (
+          <TableWithPagination
+            columns={columnNames}
+            customToolbarElement={customToolbarElement}
+            onDetailsClick={(rowData): void => {
+              setSelectedInstance(rowData as unknown as Instance);
+              setShowInstanceDrawer(true);
+            }}
+            isLoading={isLoading}
+            rows={bridgeListResponse.items}
+            totalRows={totalRows}
+            pageNumber={currentPage}
+            pageSize={currentPageSize}
+            onPaginationChange={(pageNumber, pageSize): void => {
+              const correctPageNumber =
+                pageSize === currentPageSize ? pageNumber : FIRST_PAGE;
+              setCurrentPage(correctPageNumber);
+              setCurrentPageSize(pageSize);
+              void getBridges(correctPageNumber, pageSize);
+            }}
```

Comment:
I am moving this block to a dedicated function.

I tested it a bit about the page size change near the subsequent polling execution (you can also try it locally). We have three cases:
1) Changing page size just before the polling execution will cause the polling to be executed with the current pagination parameter because we always perform the polling with the most updated version of `currentPage` and `currentPageSize`.
2) Changing page size after the polling execution will cause the "regular" call to be executed with the most updated version of `currentPage` and `currentPageSize`.
3) The "regular" call (caused by user action on pagination) and the polling calls are being executed simultaneously: only the last REST call will be completed. At the same time, the Axios Cancellation Token will cancel the old one. In any case, the "surviving" call will be executed with the updated pagination parameters.

Reply:
Thanks for clarifying the three scenarios. Like you say the correct data is always shown. I'm going to approve.

I think there's only a minor rendering issue but it's not blocking for this PR. Also because it's not easy to reproduce :). So I'm not asking to fix it here.

When the pagination call is triggered right before the polling I think this happens:
1. pagination call starts
2. skeleton data is shown
3. polling request starts
4. pagination call gets canceled
5. skeleton data is hidden, previous data is shown
6. polling request succeeds
7. new items list with correct page/size is rendered on top of the previous one. here I could think the data was already fetched at step 5.

This is nitpicking of course. I also realize this is probably noticeable mainly with the mocked apis and their high response delay (1s). Requests to the real apis are usually faster, so the overlap will be less likely to happen.

Reply:
Yes i think it's missing skeleton due to call cancellation. Not so easy to reproduce. 
I've recorded it [here](https://issues.redhat.com/secure/attachment/12717112/pagination-followed-by-polling.webm)

Reply:
I imagine that you can also reproduce with https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/commit/514842147f70940f8f9486257f61e770db726495.
I tried it locally (a lot) and still couldn't reproduce it.
However, I think a _corner_ case happens when the polling is executed right before the cancelation, so the `isLoading` flag in the hook is set to `false` because, with polling, there is no loading skeleton.

Reply:
Yes, I saw it with the latest version of the PR.
And yes my guess is the same as yours. The polling call that cancels the pagination call is also resetting the loading, I suppose [here](https://github.com/vpellegrino/sandbox-ui/blob/514842147f70940f8f9486257f61e770db726495/src/hooks/useBridgesApi/useGetBridgesApi.ts#L30) maybe.

Would you kindly open a jira to track it so we can investigate it later?